File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-java/src/test/java/com/example/statemachine/simple/EnumConditionHelper.java
package com.example.statemachine.simple;

import static org.hamcrest.Matchers.in;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import java.util.Arrays;
import org.hamcrest.Matcher;

public class EnumConditionHelper {

  public static Matcher<DocumentEvent> containsInDocumentEvents() {

    return in(DocumentEvent.values());
  }



  public static Matcher<DocumentState> containsInDocumentStates() {

    return in(DocumentState.values());
  }

  public static Matcher<String> containsInDocumentEventNames() {

    final var documentEventNameList = Arrays.stream(DocumentEvent.values())
      .map(Enum::name)
      .toList();


    return in(documentEventNameList);
  }

  public static Matcher<?> containsInDocumentStateNames() {

    final var documentStateNameList = Arrays.stream(DocumentState.values())
      .map(Enum::name)
      .toList();

    return in(documentStateNameList);
  }

}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/test/kotlin/com/example/statemachine/simple/EnumConditionHelper.kt
package com.example.statemachine.simple

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import org.hamcrest.Matcher
import org.hamcrest.Matchers
import java.util.*

object EnumConditionHelper {
    fun containsInDocumentEvents(): Matcher<DocumentEvent> {
        return Matchers.`in`(DocumentEvent.entries.toTypedArray())
    }

    fun containsInDocumentStates(): Matcher<DocumentState> {
        return Matchers.`in`(DocumentState.entries.toTypedArray())
    }

    fun containsInDocumentEventNames(): Matcher<String> {
        val documentEventNameList = Arrays.stream(DocumentEvent.entries.toTypedArray())
            .map { obj: DocumentEvent -> obj.name }
            .toList()
        return Matchers.`in`(documentEventNameList)
    }

    fun containsInDocumentStateNames(): Matcher<*> {
        val documentStateNameList = Arrays.stream(DocumentState.entries.toTypedArray())
            .map { obj: DocumentState -> obj.name }
            .toList()
        return Matchers.`in`(documentStateNameList)
    }
}
